### ISSUE TITLE
Obfuscated js from boston.com causes pywb.js to become unresponsive

### ISSUE THREAD

**Issue by rebeccacremona** [webrecorder/pywb#474]
## Describe the bug

Boston.com articles presently include the obfuscated javascript: http://dz9qn8fh4jznm.cloudfront.net/script.js. 

`pywb.js` [becomes unresponsive](https://support.mozilla.org/en-US/kb/warning-unresponsive-script) when attempting playbacks with the offending script; recordings of articles containing the offending script that are mediated by `pywb.js` (e.g., production Webrecorder + user's browser) fail.

Sample `console.log` from Firefox, which permits you to forcibly terminate ("timeout") unresponsive scripts: 
```
Error: Script terminated by timeout at:
get@https://wr.perma-archives.org/static/bundle/pywb.js?289e73ce:19:98680
T@https://wr.perma-archives.org/public/v6qj-yu8a/20190601115559js_///dz9qn8fh4jznm.cloudfront.net/script.js:14:46524
@https://wr.perma-archives.org/public/v6qj-yu8a/20190601115559js_///dz9qn8fh4jznm.cloudfront.net/script.js:14:54383
C@https://wr.perma-archives.org/public/v6qj-yu8a/20190601115559js_///dz9qn8fh4jznm.cloudfront.net/script.js:14:31917
$b@https://wr.perma-archives.org/public/v6qj-yu8a/20190601115559js_///dz9qn8fh4jznm.cloudfront.net/script.js:14:54117
@https://wr.perma-archives.org/public/v6qj-yu8a/20190601115559js_///dz9qn8fh4jznm.cloudfront.net/script.js:14:75014
C@https://wr.perma-archives.org/public/v6qj-yu8a/20190601115559js_///dz9qn8fh4jznm.cloudfront.net/script.js:14:31917
Ed@https://wr.perma-archives.org/public/v6qj-yu8a/20190601115559js_///dz9qn8fh4jznm.cloudfront.net/script.js:14:74898
Kd@https://wr.perma-archives.org/public/v6qj-yu8a/20190601115559js_///dz9qn8fh4jznm.cloudfront.net/script.js:14:77866
a$c@https://wr.perma-archives.org/public/v6qj-yu8a/20190601115559js_///dz9qn8fh4jznm.cloudfront.net/script.js:14:79050
@https://wr.perma-archives.org/public/v6qj-yu8a/20190601115559js_///dz9qn8fh4jznm.cloudfront.net/script.js:14:79725
@https://wr.perma-archives.org/public/v6qj-yu8a/20190601115559js_///dz9qn8fh4jznm.cloudfront.net/script.js:14:4
```

## Steps to reproduce the bug

- Using Google Chrome, visit webrecorder.io and attempt to record `https://www.boston.com/news/local-news/2019/06/02/7-arrested-at-anti-abortion-rally-on-boston-common`; observe that recording never proceeds
![image](https://user-images.githubusercontent.com/11020492/58808406-f53f0b80-85e7-11e9-92a8-edcba5d8fd1b.png)
- Repeat using Firefox; wait until Firefox offers to stop the script; click stop; watch the content load; stop the recording; observe that no recording was produced.
![image](https://user-images.githubusercontent.com/11020492/58808559-3e8f5b00-85e8-11e9-9bd5-88386652b47b.png)
- If desired, attempt playback of either of the two following warcs, produced by Perma, and observe similar behavior
  [Example 1](https://github.com/webrecorder/pywb/files/3247856/QSQ2-SJ6L.warc.gz)
  [Example 2](https://github.com/webrecorder/pywb/files/3247870/B6C6-7L7J.warc.gz)
- If desired, configure Webrecorder or pywb to refuse to play back `http://dz9qn8fh4jznm.cloudfront.net/script.js` if found within a warc (as with https://github.com/harvard-lil/perma/blob/develop/services/docker/webrecorder/contentcontroller.py#L401), and observe that both the example warcs above play back without problems

## Environment
observed using
- Mac OSX, Google Chrome Version 74.0.3729.169, Firefox 66.0.5
- Debian + Firefox

observed with:
- production webrecorder
- via Perma, pywb:develop as of 5/21/19 11:16AM
```
FROM webrecorder/pywb@sha256:5198abb159b1bf9af740a546a4a38e00a12f79398deb7f1d6376c318bb5f0014
```

## Additional context
Originally reported by https://github.com/harvard-lil/perma/issues/2607

**Comment by N0taN3rd** [comment]
@rebeccacremona thanks for creating this issue. 
We have identified the issue with a fix incoming. 

The offending line of code was 
```js
for (var cK, cL; cK = cI[cJ], delete cI[cJ], cL = cI[cJ], cK !== cL;);
 // where cJ === name of document native function and cI === document
```

and how we handled dynamic property accesses for `document` and `window` was causing the stopping condition to never be false (`cK !== cL`)

**Comment by rebeccacremona** [comment]
Wow! I am blown away by your debugging! 🙌 Thanks so much for taking a look!

**Comment by ikreymer** [comment]
Fixed in pywb 2.3.x!
